### PR TITLE
Add specification PODOs/models

### PR DESCRIPTION
**Pull request by Schwusch**
I've made typed classes for the json specification, which is much easier to work with than raw Map<String, dynamic>.

### CODE REVIEW

**Comment by isoos** [comment]
Nice timing! I was about to begin looking into this :)

File: generator/lib/model/api.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  final String apiVersion;
+  final String endpointPrefix;
+  final String signingName;
+  final String globalEndpoint;
+  final String signatureVersion;
+  final String jsonVersion;
+  final String targetPrefix;
+  final String protocol;
+
+  /*
```
[the remark this answers is not in the document]
Comment (Schwusch) [reply generator/lib/model/api.dart]: Good catch, don't know why the analyzer didn't remind me

File: generator/lib/model/operation.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import 'package:aws_client.generator/model/descriptor.dart';
```
[the remark this answers is not in the document]
Comment (Schwusch) [reply generator/lib/model/operation.dart]: I started doing these kind of imports (`package:my_package/foo.dart`) when the compiler had a bug(?) with relative imports. Let's say you declare a function that takes `Foo` as an argument:

```dart
import '../foo.dart';

baz(Foo) => null;
```

And another file calls this function:

```dart
import '../bar/foo.dart';

baz(Foo());
```

The compiler used to complain with something like `../bar/foo.dart is not an instance of ../foo.dart`

Is this fixed?

[the remark this answers is not in the document]
Comment (isoos) [reply generator/lib/model/operation.dart]: I haven't encountered it for more than over two years or so. It does happen if the library you are referencing is outside of `lib/`, but other then that, I'm not aware of such bug.

**Pull request merged**